File: studentgradechecker-classifygrade/src/test/java/GradeCalculatorTest.java
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.ArrayList;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import com.webservice.GradeCalculator;

public class GradeCalculatorTest {

    static String inputText;
    static String grade;
    static ArrayList<String> marks;
    static double finalMark;

    @BeforeAll
    public static void beforeAll(){
        inputText = "a,100newlineb,58newlinec,70";
        populateExpectedMarks();
    }

    @Test
    public void testMarksCorrectlyParsedFromInputText(){
        ArrayList<String> expectedMarks = marks;
        ArrayList<String> actualMarks = GradeCalculator.parseMarksFromInput(inputText);

        assertEquals(expectedMarks, actualMarks);
    }

    @Test
    public void testFinalMarkCalculatedCorrectlyFromParseMarks(){
        double expectedFinalMark = 76.0;
        double actualFinalMark = GradeCalculator.calculateFinalMarkFromMarks(marks);

        assertEquals(expectedFinalMark, actualFinalMark);
    }

    @Test
    public void testGradeCalculatedCorrectlyFromFinalMarkForDistinction(){
        Double finalMark = 70.0;
        String expectedGrade = "Distinction";
        String actualGrade = GradeCalculator.classifyGrade(finalMark);

        assertEquals(expectedGrade, actualGrade);
    }

    @Test
    public void testGradeCalculatedCorrectlyFromFinalMarkForCommendation(){
        Double finalMark = 60.0;
        String expectedGrade = "Commendation";
        String actualGrade = GradeCalculator.classifyGrade(finalMark);

        assertEquals(expectedGrade, actualGrade);
    }

    @Test
    public void testGradeCalculatedCorrectlyFromFinalMarkForPass(){
        Double finalMark = 50.0;
        String expectedGrade = "Pass";
        String actualGrade = GradeCalculator.classifyGrade(finalMark);

        assertEquals(expectedGrade, actualGrade);
    }

    @Test
    public void testGradeCalculatedCorrectlyFromFinalMarkForMarginalFail(){
        Double finalMark = 40.0;
        String expectedGrade = "Marginal Fail";
        String actualGrade = GradeCalculator.classifyGrade(finalMark);

        assertEquals(expectedGrade, actualGrade);
    }

    @Test
    public void testGradeCalculatedCorrectlyFromFinalMarkForFail(){
        Double finalMark = 30.0;
        String expectedGrade = "Fail";
        String actualGrade = GradeCalculator.classifyGrade(finalMark);

        assertEquals(expectedGrade, actualGrade);
    }

    @Test
    public void testGradeCalculatedCorrectlyFromFinalMarkForLowFail(){
        Double finalMark = 29.0;
        String expectedGrade = "Low Fail";
        String actualGrade = GradeCalculator.classifyGrade(finalMark);

        assertEquals(expectedGrade, actualGrade);
    }

    private static void populateExpectedMarks(){
        marks = new ArrayList<>();
        marks.add("100");
        marks.add("58");
        marks.add("70");
    }
    
}

File: studentgradechecker-classifyeach/src/test/kotlin/GradeCalculatorTest.kt
import com.webservice.GradeCalculator.classifyGrades
import com.webservice.GradeCalculator.parseMarksFromInput
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import java.util.*

class GradeCalculatorTest {
    
    @Test
    fun testMarksCorrectlyParsedFromInputText() {
        val inputText = "a,100newlineb,58newlinec,70"

        val marks: MutableList<Double> = ArrayList()
        marks!!.add(100.0)
        marks!!.add(58.0)
        marks!!.add(70.0)

        val expectedMarks = marks
        val actualMarks = parseMarksFromInput(inputText!!)
        Assertions.assertEquals(expectedMarks, actualMarks)
    }

    @Test
    fun testGradeCalculatedCorrectlyFromFinalMarkForDistinction() {
        val finalMark = 70.0
        val expectedGrade = "Distinction"
        val actualGrade = classifyGrades(finalMark)
        Assertions.assertEquals(expectedGrade, actualGrade)
    }

    @Test
    fun testGradeCalculatedCorrectlyFromFinalMarkForCommendation() {
        val finalMark = 60.0
        val expectedGrade = "Commendation"
        val actualGrade = classifyGrades(finalMark)
        Assertions.assertEquals(expectedGrade, actualGrade)
    }

    @Test
    fun testGradeCalculatedCorrectlyFromFinalMarkForPass() {
        val finalMark = 50.0
        val expectedGrade = "Pass"
        val actualGrade = classifyGrades(finalMark)
        Assertions.assertEquals(expectedGrade, actualGrade)
    }

    @Test
    fun testGradeCalculatedCorrectlyFromFinalMarkForMarginalFail() {
        val finalMark = 40.0
        val expectedGrade = "Marginal Fail"
        val actualGrade = classifyGrades(finalMark)
        Assertions.assertEquals(expectedGrade, actualGrade)
    }

    @Test
    fun testGradeCalculatedCorrectlyFromFinalMarkForFail() {
        val finalMark = 30.0
        val expectedGrade = "Fail"
        val actualGrade = classifyGrades(finalMark)
        Assertions.assertEquals(expectedGrade, actualGrade)
    }

    @Test
    fun testGradeCalculatedCorrectlyFromFinalMarkForLowFail() {
        val finalMark = 29.0
        val expectedGrade = "Low Fail"
        val actualGrade = classifyGrades(finalMark)
        Assertions.assertEquals(expectedGrade, actualGrade)
    }
}